react/components/SliderContext.tsx

```diff
@@ -231,12 +217,12 @@ const SliderContextProvider: FC<SliderContextProps> = ({
     }
 
     return resultingSlideWidth
-  }, [newSlides.length, centerMode, resolvedSlidesPerPage])
+  }, [newTotalItems, centerMode, resolvedSlidesPerPage])
 
   const transformMap = useMemo(() => {
     const currentMap: Record<number, number> = {}
 
-    newSlides.forEach((_, idx) => {
+    Array.from(Array(newTotalItems).keys()).forEach(idx => {
```

Comment:
hi, why not a traditional loop here?

Reply:
no particular reason

Reply:
I would add a comment here to explain this `Array.from(Array(newTotalItems).keys())`

Reply:
Yep, this is a bit difficult to understand. A comment would be great :)